### PR TITLE
[WIP] Refactor current `UserManagement` methods according to latest API reference

### PR DESCRIPTION
**Pull request by LauraBeatris**
## Description

[WIP]

## Documentation

Does this require changes to the WorkOS Docs? E.g. the [API Reference](https://workos.com/docs/reference) or code snippets need updates.

```
[X] Yes
```

If yes, link a related docs PR and add a docs maintainer as a reviewer. Their approval is required.

### CODE REVIEW

File: lib/workos/user_management.rb [hunk 1]
```diff
@@ -167,7 +167,7 @@ def create_user(email:, password: nil, first_name: nil, last_name: nil, email_ve
       end
       def update_user(id:, first_name: nil, last_name: nil, email_verified: nil)
         request = put_request(
-          path: "/users/#{id}",
+          path: "/user_management/#{id}",
```
Comment (amygdalama) [inline lib/workos/user_management.rb:170]: this (and the other user CRUD endpoints) should be `/user_management/users/`

File: lib/workos/user_management.rb [hunk 2]
```diff
@@ -513,7 +513,7 @@ def authenticate_user_with_totp(code:, client_id:, pending_authentication_token:
       def enroll_auth_factor(user_id:)
         response = execute_request(
           request: post_request(
-            path: "/users/#{user_id}/auth/factors",
+            path: "/user_management/#{user_id}/auth/factors",
```
Comment (amygdalama) [inline lib/workos/user_management.rb:516]: also `/user_management/users/`

Comment (amygdalama) [reply lib/workos/user_management.rb:516]: (and the below auth factors endpoint too)

Comment (LauraBeatris) [reply lib/workos/user_management.rb]: Good point! I was misreading the API controller, thanks for catching!

File: lib/workos/user_management.rb [hunk 3]
```diff
@@ -77,7 +77,7 @@ def delete_user(id:)
       def reset_password(token:, new_password:)
         response = execute_request(
           request: post_request(
-            path: '/users/password_reset',
+            path: '/user_management/users/password_reset',
```
Comment (amygdalama) [inline lib/workos/user_management.rb:80]: oh, not everything should have `/user_management/users`. only the REST-style endpoints for User objects... 

the general approach is:
* User Management routes are all "namespaced" under `/user_management`
* within the User Management namespace, we have some REST-style endpoints for User resources. those are `GET /user_management/users/:id` etc
* there are _also_ some endpoints that are part of the User Management product (so using the namespace), but that aren't a REST endpoint on a User resource (like sending a password reset email)

previously, we were treating `/users` as _both_ a product namespace _and_ a resource, and that's what we're moving away from.

check the spreadsheet and/or the API Reference for what these should be 👀

Comment (LauraBeatris) [reply lib/workos/user_management.rb:80]: Oh cool, I was in automatic mode and rename everywhere 😅 Going to make the fixes here.

Comment (amygdalama) [reply lib/workos/user_management.rb:80]: yeah no worries, i figured! thank youuu 🙏

**Comment by LauraBeatris** [comment]
@amygdalama Going to break this PR into a smaller one, starting to refactor methods related to `User` only https://github.com/workos/workos-ruby/pull/235

**Pull request closed without merging**